manifest-generator/generator-services/src/main/java/io/hyscale/generator/services/plugins/LoadBalancerHandler.java

```
@@ -0,0 +1,70 @@
+import io.hyscale.servicespec.commons.fields.HyscaleSpecFields;
+import io.hyscale.servicespec.commons.model.service.ServiceSpec;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+import java.util.List;
+
+@Component
+@ManifestPlugin(name = "LoadBalancerHandler")
+public class LoadBalancerHandler implements ManifestHandler {
+
+    private static final Logger logger = LoggerFactory.getLogger(LoadBalancerHandler.class);
+
+    /**
+     * @param serviceSpec
+     * @param manifestContext
+     * @return
+     * @throws HyscaleException
+     */
+    @Override
+    public List<ManifestSnippet> handle(ServiceSpec serviceSpec, ManifestContext manifestContext) throws HyscaleException {
+        if (!ManifestPredicates.getLoadBalancerPredicate(LBType.INGRESS).test(serviceSpec) && !ManifestPredicates.getLoadBalancerPredicate(LBType.ISTIO).test(serviceSpec)) {
+            logger.debug("Load Balancer information found to be empty while processing service spec data.");
+            return Collections.emptyList();
+        }
+        LoadBalancer loadBalancer = serviceSpec.get(HyscaleSpecFields.loadBalancer, new TypeReference<LoadBalancer>() {
+        });
+        try {
+            LBType lbType = LBType.getByProvider(loadBalancer.getProvider());
+            ServiceMetadata serviceMetadata = new ServiceMetadata();
+            serviceMetadata.setAppName(manifestContext.getAppName());
+            serviceMetadata.setEnvName(manifestContext.getEnvName());
+            String serviceName = serviceSpec.get(HyscaleSpecFields.name, String.class);
+            serviceMetadata.setServiceName(serviceName);
+            return lbType.getBuilder().build(serviceMetadata, loadBalancer);
+        } catch (HyscaleException e) {
+            logger.error("Error while generating LoadBalancer manifests.", e);
+            return Collections.emptyList();
```

Better to rethrow the exception after logging instead of returning empty list

---

If we rethrow an exception after logging it, This causes a sonar violation.